codec/types/any.go

```diff
@@ -58,15 +58,16 @@ type Any struct {
 // returns an error if that value couldn't be packed. This also caches
 // the packed value so that it can be retrieved from GetCachedValue without
 // unmarshaling
-func NewAnyWithValue(value proto.Message) (*Any, error) {
-	any := &Any{}
-
-	err := any.Pack(value)
-	if err != nil {
-		return nil, err
+func NewAnyWithValue(v proto.Message) (*Any, error) {
+	if v == nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrPackAny, "Expecting non nil value to create a new Any")
 	}
-
-	return any, nil
+	bz, err := proto.Marshal(v)
+	return &Any{
+		TypeUrl:     "/" + proto.MessageName(v),
```

Comment:
can this call NewAnyWithCustomTypeURL with proto.MessageName(v)?

Reply:
Yes, good idea.